### ISSUE TITLE
Non-English keyboards are not mapped correctly on Macbook

### ISSUE THREAD

**Issue by neuos** [utmapp/UTM#2678]
Not all keys work correctly if the physical keyboard is non-English. Eg. on a German keyboard all the keys work normally except the <> key prints ^°  and the ^° doesn't fire any event (checked with xev in Ubuntu).

![image](https://user-images.githubusercontent.com/25673607/126565520-3c1b015c-21b5-47c0-8588-e311b6551c82.png)

This behaviour was observed on an Ubuntu 20.04 vm as well as on a Windows 10 vm.

UTM Version 2.1.2 (31)

**Comment by nqvst** [comment]
additionally the right alt key is reporting the same keycode as the left alt key

**Comment by LordBongio** [comment]
Same for italian keyboard, third layout of keys doesn’t work and same keys as the image don’t fire

**Comment by speetroo** [comment]
I have an M1 Macbook with a French Keyboard and some keys aren't working properly; >< are showing @ ; and so @ is not working at all, also, the command key (right or left) is not working -> I can't use any of the special keys like option + ) for } or shift + option + ) for ].

Everything else is working fine for me, unfortunately the keyboard issue is kind of killing my experience

**Comment by osy** [comment]
Are you choosing the right keyboard layout in your guest VM?

**Comment by speetroo** [comment]
Yes, the selected keyboard is "French (Macintosh)" and when I show it it is exactly the same keys as my keyboard, and every keys are working properly when I test them (it becomes grey on the keyboard previsualisation on ubuntu) except the (important) ones I talked about in my previous message :/

**Comment by osy** [comment]
The emulated keyboard shouldn’t be Macintosh. Is there an option for ANSI or ISO?

**Comment by speetroo** [comment]
Do you mean in ubuntu input source keyboard in country and langage ? If yes not I can't seem to find it, but all this would be so much simpler if I could directly show you, maybe you have a discord ?

**Comment by osy** [comment]
https://discord.gg/UV2RUgD

**Comment by antofic** [comment]
Hello, I'm testing UTM 3.1.5 and UTM 3.2.3 beta with Mac OS 9.1.2 on an intel iMac and a silicon Mac Studio and an Apple Italian 
keyboard with numeric keypad.

I have selected the Italian-Pro keyboard in macOS.
The '=' key on the numeric keypad does not work and, consequently, does not work even when paired with modifier keys.
For example, menus that require 'command' + '=' don't work.
All other keys appear to be working correctly.

The '=' key does not work even when selecting the 'Italian' keyboard.

Vice versa 'shift' + '0' on the non-numeric part of the keyboard correctly generates '='.

Sorry for my bad English.

**Comment by menzo** [comment]
I have the same issue with UTM 3.2.4 and a MacBook with hungarian keyboard. I've installed a preconfigured macOS 12+ on Monterey host. I tried to switch the emulated keyboard to ISO, but no luck.

**Comment by rgayraud** [comment]
Same problem here with french keyboard on Macbook Air M2, with UTM 4.2.5 and Apple Virtualization. This is marked as a "closed" issue. Is there a possibility to reopen it ?

**Comment by osy** [comment]
Please open a new issue with the information requested.

**Comment by msander** [comment]
See my issue and naga-re's response. Seems to be Apple's responsibility; maybe we can create some kind of workaround.... https://github.com/utmapp/UTM/issues/5037#issuecomment-1440699984
